src/wp-includes/class-wp-query.php

```diff
@@ -3476,8 +3476,27 @@ public function get_posts() {
 			$this->posts = array_map( 'get_post', $this->posts );
 
 			if ( $q['cache_results'] ) {
-				$post_ids = wp_list_pluck( $this->posts, 'ID' );
-				_prime_post_caches( $post_ids, $q['update_post_term_cache'], $q['update_post_meta_cache'] );
+				$post_ids       = wp_list_pluck( $this->posts, 'ID' );
+				$non_cached_ids = _get_non_cached_ids( $post_ids, 'posts' );
+				if ( ! empty( $non_cached_ids ) ) {
+					$fresh_posts = array();
+					foreach ( $this->posts as $post ) {
+						if ( in_array( $post->ID, $non_cached_ids, true ) ) {
+							$fresh_posts[] = $post;
+						}
+					}
+					update_post_cache( $fresh_posts );
+				}
```

Comment:
Is updating the cache so expensive as to warrant this? If it's not a significant perf hit this can probably be simplified to `update_post_caches( $this-posts, 'any', $q['update_post_term_cache'], $q['update_post_meta_cache'] );`.

Untested but you see the point.

Reply:
I think we should switch back to `update_post_caches`.  See the original discussion. https://github.com/WordPress/wordpress-develop/pull/3348#discussion_r981797850

The issue is, setting the cache for post that hasn't changed on EVERY request, seems very pointless and wasteful, if you are using an external object cache.

Reply:
I am not against using `update_post_caches`. This is a revert and we add this to 6.1.2 and work out a better solution for 6.2